qasc/evaluator/publish_for_leaderboard.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+# This script will test the evaluator, build a docker image, and publish it as
+# a Beaker image owned by the Leaderboard user. This is meant to be run by AI2
+# after making changes to the QASC evaluator.
+
+set -e
+
+echo --------------------
+echo Unit tests
+echo --------------------
+echo
+
+set -x
+python3 test_evaluator.py
+set +x
+
+echo
+echo --------------------
+echo Test docker image
+echo --------------------
+echo
+
+set -x
+./test.sh
+set +x
+
+echo
+echo --------------------
+echo Build local docker image
+echo --------------------
+echo
+
+NAME="qasc-evaluator-$(date +"%Y%m%d-%H%M%S")"
+
+set -x
+docker build -t $NAME .
+set +x
+
+echo
+echo --------------------
+echo Publish Beaker image
+echo --------------------
+echo
+
+# Beaker must be configured to run as the leaderboard user.
+cat >>/tmp/beaker-leaderboard-config.yml <<EOF
+agent_address: https://beaker.org
+user_token: $(vault read -field=token secret/ai2/alexandria/beaker/dev)
+EOF
+
+set -x
+export BEAKER_CONFIG_FILE=/tmp/beaker-leaderboard-config.yml
```

Comment:
I am not sure but will this variable value still be set after we exit the script ? If so, can you unset this at the end, so we don't  run leaderboard experiments after publishing this ?

Reply:
Good question. It won't leak.

```
michalg  ~ $ cat test.sh
#!/bin/bash
export FOO=123
echo FOO is $FOO
michalg  ~ $ ./test.sh  
FOO is 123
michalg  ~ $ echo FOO is $FOO
FOO is
```